stonesoup/updater/particle.py

```diff
@@ -44,29 +43,33 @@ def update(self, hypothesis, **kwargs):
         : :class:`~.ParticleState`
             The state posterior
         """
-        particles = copy.copy(hypothesis.prediction.particles)
+        predicted_state = hypothesis.prediction
```

Comment:
Without copying, this is modifying the predicted state itself.

```suggestion
        predicted_state = copy.copy(hypothesis.prediction)
```